digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/ProjectService.scala

```diff
@@ -531,6 +531,27 @@ class ProjectService(roadAddressService: RoadAddressService, roadLinkService: Ro
         discontinuity = discontinuity.getOrElse(Continuous))
     }
 
+    /**
+      * Fetches the projectLink name, first from the project link, if that's not available then search for the road address.
+      *
+      * @param projectLink
+      * @return
+      */
+    def fillRoadNames(projectLink: ProjectLink): ProjectLink = {
+      val projectLinkName = ProjectLinkNameDAO.get(projectLink.roadNumber, projectLink.projectId)
+      val trueRoadName = if (projectLinkName.isEmpty) {
+        val roadName = RoadNameDAO.getLatestRoadName(projectLink.roadNumber)
+        if (roadName.isEmpty) {
```

Comment:
I'm sure we have a better way to validate this and reduce the amount of code just to make sure that roadname is not empty. You can use headOption or  .map(_.roadName).getOrElse(""). All this "if else if else" code can be suspicious in the future